init/setup-kdump.functions

```diff
@@ -715,6 +715,44 @@ function kdump_get_targets()						   # {{{
     return 0
 }									   # }}}
 
+#
+# Get root device from the kernel command line.
+#
+# Output:
+#   string that contains the device name
+function kdump_get_root_from_commandline()
+{
+    local _o _val
+    unset _val
+    unset _o
+    CMDLINE=$(cat /proc/cmdline)
+    _arg="root"
+
+    for _o in $CMDLINE; do
+        if [ "${_o%%=*}" = "${_arg%%=*}" ]; then
+            if [ -n "${_arg#*=}" -a "${_arg#*=*}" != "${_arg}" ]; then
+                # if $1 has a "=<value>", we want the exact match
+                if [ "$_o" = "_arg" ]; then
+                    _val="1";
+                fi
+                continue
+            fi
+
+            if [ "${_o#*=}" = "$_o" ]; then
```

Comment:
This is also bogus. Although the kernel arguments without value, the `root` argument is recognized only if it has a value. Take this (evil) command line:
```
root=/dev/sda1 root
```
The Linux kernel will disregard the second `root` argument, but your code will parse it and set `_val="1"`